lighthouse-core/gather/gatherers/dobetterweb/notification-on-start.js

```diff
@@ -0,0 +1,67 @@
+/**
+ * @license
+ * Copyright 2016 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Captures calls to the notification API on page load.
+ */
+
+'use strict';
+
+const Gatherer = require('../gatherer');
+
+/* global navigator */
+
+/* istanbul ignore next */
+function queryNotificationPermission() {
```

Comment:
yeah, it would be great to pull this out into `driver.queryPermissionStatus` or whatever